src/js/components/CosmosErrorMessage.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class CosmosErrorMessage extends React.Component {
+  getHeader() {
+    let {header, headerClass} = this.props;
+    if (!header) {
+      return null;
+    }
+
+    return (
+      <span className={headerClass}>
+        {header}
+      </span>
+    );
+  }
+  getMessage() {
+    let {type, message} = this.props.error;
+
+    switch (type) {
+      // Specific uninstall errors
+      case 'IncompleteUninstall':
+      case 'UninstallNonExistentAppForPackage':
+        if (!message) {
+          message = 'Unable to uninstall package. Please check your connection and try again.';
+        }
+        break;
+
+      // // Specific install errors
+      case 'PackageAlreadyInstalled':
+      case 'PackageNotInstalled':
+      case 'PackageNotFound':
+      case 'VersionNotFound':
+        if (!message) {
+          message = 'Unable to install package. Please check configuration of the package and try again.';
+        }
+        break;
+
+      // Repository related errors that can occur at any request to Cosmos
+      case 'EmptyPackageImport':
+      case 'IndexNotFound':
+      case 'PackageFileMissing':
+      case 'PackageFileNotJson':
+      case 'RepositoryNotPresent':
+      case 'RepositoryUriConnection':
+      case 'RepositoryUriSyntax':
+        if (!message) {
+          message = 'The URL for a repository is not valid, or its host did not resolve. You might need to change the URL of this repository.';
+        }
+        message = this.appendRepositoryLink(message);
+        break;
+    }
+
+    return message;
```

Comment:
what do you think about prepending the `type` to the message? like how javascript does it

`TypeError: 'forEach' is not a function...'`